cli/tests/test_tag_command.rs

```diff
@@ -77,3 +79,70 @@ fn test_tag_list() {
         test_tag2
          "###);
 }
+
+#[test]
+fn test_tag_forget() {
+    let test_env = TestEnvironment::default();
+    test_env.add_config("git.auto-local-branch = true");
+    let git_repo_path = test_env.env_root().join("source");
+    let git_repo = git2::Repository::init(git_repo_path).unwrap();
+
+    set_up_tagged_git_repo(&git_repo);
+
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "clone", "source", "tagged"]);
```

Comment:
can you add colocated test?

I expect it wouldn't work because forgotten tags aren't exported.

Reply:
Good point! I guess I should see about exporting tag deletion in this case.

Reply:
> that might cause tag deletion on a git push

You mean `git push` (not `jj git push`)?

We don't have `remote_views["git"].tags` yet, but I think exporting forgotten tags is okay.

Reply:
Yes, I mean the Git `git push --tags`. There is no problem if we're talking about tags that are already deleted on the remote, but I'm worried that if they aren't, Git will delete them on the remote on push.

We could try to delete remote-tracking Git tags, if there is such a thing.

If there is such a think as remote-tracking tags, another alternative would be to only allow forgetting a tag if no "remote-tracking tags" exist in Git.

Every way seems messy.

Reply:
Tag handling in Git is peculiar. I don't remember how to delete remote tags by git, but we'll need `--delete` or something?
https://git-scm.com/book/en/v2/Git-Basics-Tagging#_deleting_tags